tests/test_datasets.py

```diff
@@ -486,206 +553,272 @@ def test_cli_dataset_put_feature_fromfile(tmpdir):
+    assert json.loads(responses.calls[0].request.body) == {"put": features}
 
-    assert result.exit_code == 0
-    assert result.output.strip() == feature.strip()
 
 @responses.activate
-def test_cli_dataset_delete_feature():
-    dataset = "dataset-2"
-    id = "abc"
+def test_datasets_batch_write_features_exception():
+    dataset = 'abc'
 
     responses.add(
-        responses.DELETE,
-        'https://api.mapbox.com/datasets/v1/{0}/{1}/features/{2}?access_token={3}'.format(username, dataset, id, access_token),
+        responses.POST,
+        'https://api.mapbox.com/datasets/v1/{0}/{1}/features?access_token={2}'.format(username, dataset, access_token),
         match_querystring=True,
-        status=204
+        body=json.dumps({"Message": "humbug"}),
+        content_type='application/json',
+        status=401
     )
 
-    runner = CliRunner()
-    result = runner.invoke(
-        main_group,
-        ['--access-token', access_token,
-         'datasets',
-         'delete-feature', dataset, id])
+    try:
+        datasets.batch_write_features(service, dataset, ["bah"])
+    except MapboxCLIException:
+        assert True
+    else:
+        assert False
 
-    assert result.exit_code == 0
 
 @responses.activate
-def test_cli_dataset_batch_update_feature_inline():
-    dataset = "dataset-3"
-    puts = """
-    [
-      {"type":"Feature","id":"a","properties":{},"geometry":{"type":"Point","coordinates":[0,0]}},
-      {"type":"Feature","id":"b","properties":{},"geometry":{"type":"Point","coordinates":[1,1]}}
-    ]
-    """
-    deletes = '["c"]'
-    puts = "".join(puts.split())
-    expected='{{"put":{0},"delete":{1}}}'.format(puts, deletes)
+def test_datasets_write_features_append_to_dataset():
+    dataset = 'abc'
+    features = []
+    for i in range(1, 124):
+        features.append({
+            "type": "Feature",
+            "properties": {},
+            "geometry": {
+                "type": "Point",
+                "coordinates": [180 % i, 90 % i]
+            }
+        })
 
     responses.add(
         responses.POST,
         'https://api.mapbox.com/datasets/v1/{0}/{1}/features?access_token={2}'.format(username, dataset, access_token),
-        status=200, body=expected,
         match_querystring=True,
+        body=json.dumps({"put": [], "delete": []}),
         content_type='application/json'
     )
 
-    runner = CliRunner()
-    result = runner.invoke(
-        main_group,
-        ['--access-token', access_token,
-         'datasets',
-         'batch-update-features', dataset, puts, deletes])
+    datasets.write_features(
+        features,
+        True,
+        None,
+        None,
+        service,
+        'mapbox://datasets/{0}/{1}'.format(username, dataset)
+    )
 
-    assert result.exit_code == 0
-    assert result.output.strip() == expected
+    assert len(responses.calls) == 2
+    for call in responses.calls:
+        assert call.request.method == 'POST'
 
```

Comment:
Delete assertions on the mock responses, restore assertions on commands exit code and output.